commit 93a575ab15332e2037d35744c50412b40e6511a4
Author: Daniele Ceraolo Spurio <daniele.ceraolospurio@intel.com>
Date:   Wed May 31 16:54:10 2023 -0700

    drm/i915/huc: Parse the GSC-enabled HuC binary
    
    The new binaries that support the 2-step authentication contain the
    legacy-style binary, which we can use for loading the HuC via DMA. To
    find out where this is located in the image, we need to parse the
    manifest of the GSC-enabled HuC binary. The manifest consist of a
    partition header followed by entries, one of which contains the offset
    we're looking for.
    Note that the DG2 GSC binary contains entries with the same names, but
    it doesn't contain a full legacy binary, so we need to skip assigning
    the dma offset in that case (which we can do by checking the ccs).
    Also, since we're now parsing the entries, we can extract the HuC
    version that way instead of using hardcoded offsets.
    
    Note that the GSC binary uses the same structures in its binary header,
    so they've been added in their own header file.
    
    v2: fix structure names to match meu defines (s/CPT/CPD/), update commit
        message, check ccs validity, drop old version location defines.
    
    v3: drop references to the MEU tool to reduce confusion, fix log (John)
    
    v4: fix log for real (John)
    
    Signed-off-by: Daniele Ceraolo Spurio <daniele.ceraolospurio@intel.com>
    Cc: Alan Previn <alan.previn.teres.alexis@intel.com>
    Cc: John Harrison <John.C.Harrison@Intel.com>
    Reviewed-by: Alan Previn <alan.previn.teres.alexis@intel.com> #v2
    Reviewed-by: John Harrison <John.C.Harrison@Intel.com>
    Link: https://patchwork.freedesktop.org/patch/msgid/20230531235415.1467475-3-daniele.ceraolospurio@intel.com

diff --git a/drivers/gpu/drm/i915/gt/uc/intel_huc_fw.h b/drivers/gpu/drm/i915/gt/uc/intel_huc_fw.h
index db42e238b45f..0999ffe6f962 100644
--- a/drivers/gpu/drm/i915/gt/uc/intel_huc_fw.h
+++ b/drivers/gpu/drm/i915/gt/uc/intel_huc_fw.h
@@ -7,8 +7,11 @@
 #define _INTEL_HUC_FW_H_
 
 struct intel_huc;
+struct intel_uc_fw;
+
+#include <linux/types.h>
 
 int intel_huc_fw_load_and_auth_via_gsc(struct intel_huc *huc);
 int intel_huc_fw_upload(struct intel_huc *huc);
-
+int intel_huc_fw_get_binary_info(struct intel_uc_fw *huc_fw, const void *data, size_t size);
 #endif